Bond/Bond.swift

```diff
@@ -66,8 +67,8 @@ public class Bond<T> {
   }
   
   public func bind(dynamic: Dynamic<T>, fire: Bool, strongly: Bool) {
-    dynamic.bonds.append(BondBox(self))
-    
+    dynamic.bonds.insert(BondBox(self))
+
     if strongly {
       self.bondedDynamics.append(dynamic)
```

Comment:
I think a counted set is best fit for all reference counting tasks! 

But since the only way one can manipulate these arrays is `bind(...)` and `unbindAll()`, I think an array is OK! Maybe it could cause trouble if there was an individual `unbind()` operation – for example, if user could do so:
```
b.bind(a) // 1st time
<...>
b.bind(a) // 2nd time
<...>
b.unbind(a) // binding dangling
```
But so far it seems this is out of scope of this issue / PR.